src/api/routes/webhooks/#webhook_id/#token/index.ts

```diff
@@ -75,16 +77,524 @@ const messageUpload = multer({
+	const limits = Config.get().limits;
+	if (limits.absoluteRate.register.enabled) {
+		const count = await Message.count({
+			where: {
+				channel_id: webhook.channel_id,
+				timestamp: MoreThan(
+					new Date(
+						Date.now() - limits.absoluteRate.sendMessage.window,
+					),
+				),
+			},
+		});
+
+		if (count >= limits.absoluteRate.sendMessage.limit)
+			throw FieldErrors({
+				channel_id: {
+					code: "TOO_MANY_MESSAGES",
+					message: req.t("common:toomany.MESSAGE"),
+				},
+			});
+	}
+
+	const files = (req.files as Express.Multer.File[]) ?? [];
+	for (const currFile of files) {
+		try {
+			const file = await uploadFile(
+				`/attachments/${webhook.channel.id}`,
+				currFile,
+			);
+			attachments.push(
+				Attachment.create({ ...file, proxy_url: file.url }),
+			);
+		} catch (error) {
+			return res.status(400).json({ message: error?.toString() });
+		}
+	}
+
+	const embeds = body.embeds || [];
+	const message = await handleMessage({
+		...body,
+		username: body.username || webhook.name,
+		avatar_url: body.avatar_url || webhook.avatar,
+		type: 0,
+		pinned: false,
+		webhook_id: webhook.id,
+		application_id: webhook.application?.id,
+		embeds,
+		// TODO: Support thread_id/thread_name once threads are implemented
+		channel_id: webhook.channel_id,
+		attachments,
+		timestamp: new Date(),
+	});
+
+	// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+	//@ts-ignore dont care2
+	message.edited_timestamp = null;
+
+	webhook.channel.last_message_id = message.id;
+
+	await Promise.all([
+		message.save(),
+		webhook.channel.save(),
+		emitEvent({
+			event: "MESSAGE_CREATE",
+			channel_id: webhook.channel_id,
+			data: message,
+		} as MessageCreateEvent),
+	]);
+
+	// no await as it shouldnt block the message send function and silently catch error
+	postHandleMessage(message).catch((e) =>
+		console.error("[Message] post-message handler failed", e),
+	);
+
+	return res.json(message);
+};
+
+const parseGitHubWebhook = (
+	req: Request,
+	res: Response,
+	next: NextFunction,
+) => {
+	const eventType = req.headers["x-github-event"] as string;
+	if (!eventType) {
+		throw new HTTPError("Missing X-GitHub-Event header", 400);
+	}
+
+	if (eventType === "ping") {
+		return res.status(200).json({ message: "pong" });
+	}
+
+	const discordPayload = transformGitHubToDiscord(eventType, req.body);
+	if (!discordPayload) {
+		// Unsupported event type
+		return res.status(204).send();
+	}
+
+	req.body = discordPayload;
+	console.dir(req.body, { depth: null });
+	// Set default wait=true for GitHub webhooks so they get a response
+	req.query.wait = req.query.wait || "true";
+	next();
+};
+
+function transformGitHubToDiscord(
+	eventType: string,
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	payload: any,
+): WebhookExecuteSchema | null {
+	switch (eventType) {
+		case "star":
+			return {
+				username: "GitHub",
+				// TODO: Provide a static avatar for GitHub
```

Comment:
Not sure what it would take, but having the GitHub logo as a URL to reference here would be nice. This implementation leaves the avatar blank.